src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/DirectionUploadOrderAboutToSubmitHandler.java

```diff
@@ -106,78 +111,90 @@ private void handleNewDocumentInUnprocessedApprovedDocuments(FinremCaseData case
+    private Map<String, CaseDocument> getStampedDocuments(FinremCaseData caseData, String userAuthorisation, String caseId) {
+        Map<String, CaseDocument> stampedDocuments = new HashMap<>();
+        StampType stampType = documentHelper.getStampType(caseData);
+        String documentCategoryId = DocumentCategory.APPROVED_ORDERS.getDocumentCategoryId();
+        List<DirectionOrderCollection> unprocessedApprovedDocuments = nullSafeUnprocessedApprovedDocuments(caseData);
+
+        unprocessedApprovedDocuments.forEach(doc -> {
+            if (!isNewDocument(doc)) {
+                CaseDocument originalDocument = doc.getValue().getOriginalDocument();
+                CaseDocument uploadedDocument = doc.getValue().getUploadDraftDocument();
+                CaseDocument stampedDocument = genericDocumentService.stampDocument(uploadedDocument, userAuthorisation,
+                    stampType, caseId);
+                stampedDocument.setCategoryId(documentCategoryId);
+
+                stampedDocuments.put(originalDocument.getDocumentUrl(), stampedDocument);
+            }
+        });
 
+        return stampedDocuments;
+    }
+
+    private void handleDraftOrderDocuments(FinremCaseData caseData, Map<String, CaseDocument> stampedDocuments) {
         List<DraftOrderDocReviewCollection> collector = new ArrayList<>();
+
         hasApprovableCollectionReader.filterAndCollectDraftOrderDocs(caseData.getDraftOrdersWrapper().getDraftOrdersReviewCollection(),
             collector, APPROVED_BY_JUDGE::equals);
 
         getApprovedDocumentsToProcess(caseData)
             .forEach(unprocessedApprovedOrder ->
-                collector.stream().filter(psa -> doesDocumentMatch(psa, unprocessedApprovedOrder)).forEach(toBeUpdated -> {
-
-                    StampType stampType = documentHelper.getStampType(caseData);
-                    CaseDocument stampedDoc = genericDocumentService.stampDocument(
-                        unprocessedApprovedOrder.getValue().getUploadDraftDocument(), authorisation, stampType,
-                        String.valueOf(caseDetails.getId()));
-                    stampedDoc.setCategoryId(
-                        DocumentCategory.APPROVED_ORDERS.getDocumentCategoryId());
-
-                    toBeUpdated.getValue().setOrderStatus(PROCESSED);
-                    toBeUpdated.getValue().setDraftOrderDocument(stampedDoc);
-                }));
+                collector.stream()
+                    .filter(psa -> doesDocumentMatch(psa, unprocessedApprovedOrder))
+                    .forEach(toBeUpdated -> {
+                        CaseDocument originalDocument = toBeUpdated.getValue().getDraftOrderDocument();
+                        CaseDocument stampedDocument = stampedDocuments.get(originalDocument.getDocumentUrl());
+                        toBeUpdated.getValue().setOrderStatus(PROCESSED);
+                        toBeUpdated.getValue().setDraftOrderDocument(stampedDocument);
```

Comment:
add `.map(XXXXX::getValue)` to so we don't need to call `getValue()` all the time.

Reply:
Done